File: 1679-max-number-of-k-sum-pairs/1679-max-number-of-k-sum-pairs.ts
function maxOperations(nums: number[], k: number): number {
      nums.sort((a,b) => a - b);
        let i = 0, j = nums.length - 1;
        let count = 0;
        
        while(i < j){
            if(nums[i] + nums[j] == k){
                count++;
                i++;
                j--;
            } 
            else if(nums[i] + nums[j] < k) i++;
            else j--;
        }
        return count;
};

File: 1679-max-number-of-k-sum-pairs/1679-max-number-of-k-sum-pairs.py
class Solution:
    def maxOperations(self, nums: List[int], k: int) -> int:
        i = 0
        j = len(nums) - 1
        count = 0
        
        nums.sort()
        
        while i < j:
            if(nums[i] + nums[j] == k):
                count += 1
                i += 1
                j -= 1
                
            elif(nums[i] + nums[j] < k):
                i += 1
                
            else: j -= 1
                
        return count